packages/dds/tree/src/feature-libraries/editable-tree/editableTree.ts

```diff
@@ -368,6 +380,10 @@ class NodeProxyTarget extends ProxyTarget<Anchor> {
         return proxifiedField;
     }
 
+    public getWithoutUnwrapping(fieldKey: FieldKey): EditableField {
+        return this.proxifyField(fieldKey, false) as EditableField;
```

Comment:
The need for a cast creates a bug/maintainability risk. It looks like we would need to refactor `proxifyField` (in two methods) in order to avoid the cast which in turns requires more refactors. All of that refactoring would belong in a separate PR (that would be merged before this one). Is there some blocking factor that would prevent the refactor (which may also be the reason why the code is factored this way in the first place)?

Reply:
There is no blockers. I have done that already in editing, so just copy it here.

Reply:
Oh I didn't realize that `EditableField` was a possible type for `UnwrappedEditableField`. That's quite confusing but that's not really this PR's problem to address that.